general/mem_manager.hpp

```diff
@@ -907,13 +908,30 @@ inline void Memory<T>::MakeAlias(const Memory &base, int offset, int size)
    h_mt = base.h_mt;
    h_ptr = base.h_ptr + offset;
    if (!(base.flags & REGISTERED))
-   { flags = (base.flags | ALIAS) & ~(OWNS_HOST | OWNS_DEVICE); }
-   else
    {
-      const size_t s_bytes = size*sizeof(T);
-      const size_t o_bytes = offset*sizeof(T);
-      MemoryManager::Alias_(base.h_ptr, o_bytes, s_bytes, base.flags, flags);
+      // TODO: Always registering 'base' (if MemoryManager::Exists()) seems to
+      // create issues in some unit tests with errors:
+      //    "alias already exists with different base/offset!"
+      // which is probably due to dangling aliases.
```

Comment:
So in the unit tests, dangling aliases can be cought?

Reply:
I'm not sure the issue is really due to dangling aliases but, yes, some of the tests (unit and other) fail with the message "alias already exists with different base/offset!" when I try to always register the 'base'.

We can investigate (and fix) this further now, or leave it for later. What do you think?